packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -942,6 +959,16 @@ class _WrappingOverlayState extends State<_WrappingOverlay> {
   }
 }
 
+class _OverlayScope extends InheritedWidget {
```

Comment:
I think there's already the `_RenderTheaterMarker` widget that can give you the `OverlayState`?